src/codeManager.ts

```diff
@@ -92,7 +100,15 @@ export class CodeManager implements vscode.Disposable {
         this._appInsightsClient.sendEvent("runByLanguage");
         const config = this.getConfiguration();
         const executorMap = config.get<any>("executorMap");
-        vscode.window.showQuickPick(Object.keys(executorMap), { placeHolder: "Type or select language to run" }).then((languageId) => {
+
+        let languages = Object.keys(executorMap);
+
+        const firstLineInFile = vscode.window.activeTextEditor.document.lineAt(0).text;
+        if (firstLineInFile.startsWith("#!")) {
+            languages = [firstLineInFile.substr(2)].concat(languages);

```

Comment:
Will this work? `languages` is array of language id, while `firstLineInFile.substr(2)` is the actual executor

Reply:
I think we could just keep `runByLanguage()` as origin. What's your idea? 😃

Reply:
Yes, it works. For me actually _languages_ is an array of strings. So you want just to use it as an origin and push that value to it?

Reply:
Actually, it it not working :)  Since `runByLanguage()` will always call `run()`, no matter what you select in the dropdown list, it will always trigger shebang command.

Reply:
So let's keep it as origin.

Reply:
You're right! I should check more careful there.